diff --git a/include/linux/sys.h b/include/linux/sys.h
index c538fc1..0e1729f 100644
--- a/include/linux/sys.h
+++ b/include/linux/sys.h
@@ -70,6 +70,7 @@ extern int sys_sgetmask();
 extern int sys_ssetmask();
 extern int sys_setreuid();
 extern int sys_setregid();
+extern int sys_hello();
 
 fn_ptr sys_call_table[] = { sys_setup, sys_exit, sys_fork, sys_read,
 sys_write, sys_open, sys_close, sys_waitpid, sys_creat, sys_link,
@@ -83,4 +84,4 @@ sys_getgid, sys_signal, sys_geteuid, sys_getegid, sys_acct, sys_phys,
 sys_lock, sys_ioctl, sys_fcntl, sys_mpx, sys_setpgid, sys_ulimit,
 sys_uname, sys_umask, sys_chroot, sys_ustat, sys_dup2, sys_getppid,
 sys_getpgrp, sys_setsid, sys_sigaction, sys_sgetmask, sys_ssetmask,
-sys_setreuid,sys_setregid };
+sys_setreuid,sys_setregid, sys_hello };
diff --git a/include/unistd.h b/include/unistd.h
index 956554a..06c36d8 100644
--- a/include/unistd.h
+++ b/include/unistd.h
@@ -129,6 +129,7 @@
 #define __NR_ssetmask	69
 #define __NR_setreuid	70
 #define __NR_setregid	71
+#define __NR_hello	72
 
 #define _syscall0(type,name) \
   type name(void) \
@@ -250,5 +251,6 @@ int dup2(int oldfd, int newfd);
 int getppid(void);
 pid_t getpgrp(void);
 pid_t setsid(void);
+int hello(void);
 
 #endif
diff --git a/kernel/sys.c b/kernel/sys.c
index 2c01e67..1c61050 100644
--- a/kernel/sys.c
+++ b/kernel/sys.c
@@ -234,3 +234,10 @@ int sys_umask(int mask)
 	current->umask = mask & 0777;
 	return (old);
 }
+
+int sys_hello(void)
+{
+    printk("Hello, Linux 0.11\n");
+
+    return 0;
+}
diff --git a/kernel/system_call.s b/kernel/system_call.s
index c602cd8..fc85e3e 100644
--- a/kernel/system_call.s
+++ b/kernel/system_call.s
@@ -58,7 +58,7 @@ sa_mask = 4
 sa_flags = 8
 sa_restorer = 12
 
-nr_system_calls = 72
+nr_system_calls = 73
 
 /*
  * Ok, I get parallel printer interrupts while using the floppy for some
